### ISSUE TITLE
Implicit namespace packages (PEP420) cause import-error

### ISSUE THREAD

**Issue by csymeonides-mf** [PyCQA/pylint#4057]
**Is this fixed in the preview release?** No

### Steps to reproduce
I'm experiencing the same issue as #842.

Trying to use PEP420 Implicit Namespace Packages (i.e. without an `__init__.py` in the top-level package directory). My tree looks like this:
```
src/
    my_namespace/
        my_package/
            __init__.py
            foo.py
            bar.py
```

and `foo.py` contains:
```
from my_namespace.my_package.bar import Bar
```

### Current behavior
Running `pylint src` works fine, but `pylint src/my_namespace/my_package/foo.py` returns an `import-error` which is a false positive.

### Expected behavior
No `import-error`

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
```

**Comment by ssfabiopicchi** [comment]
I second this one. I set up a simple repo to reproduce the issue:
https://github.com/ssfabiopicchi/precommit-test

```
pylint 2.6.0
astroid 2.4.2
Python 3.8.6 | packaged by conda-forge | (default, Oct  7 2020, 19:08:05) 
[GCC 7.5.0]
```

Somehow it works for me in another repo, but I haven't investigated it further yet.

**Comment by ssfabiopicchi** [comment]
At first I thought the issue was related to `pre-commit` because it was only happening when running with pre-commit and not when running it directly

**Comment by bdegreve** [comment]
Is the package properly installed?  Taken at face value, the top namespace would have to be `src`, not `my_namespace`.  pylint may not know in needs to start searching in `src`.  When installing it with following setup.py, I don't have any troubles:

```
from setuptools import find_namespace_packages, setup
setup(
    name="my-package",
    version="1.0.0",
    packages=["my_namespace.my_package"],
    package_dir={
        '': 'src',
    },
)
```

Then install with:
```
pip install --editable .
```

Then:
```
pylint .\src\my_namespace\my_package\foo.py
************* Module my_package.foo
src\my_namespace\my_package\foo.py:1:0: C0304: Final newline missing (missing-final-newline)
src\my_namespace\my_package\foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
src\my_namespace\my_package\foo.py:1:0: C0114: Missing module docstring (missing-module-docstring)
src\my_namespace\my_package\foo.py:1:0: W0611: Unused Bar imported from my_namespace.my_package.bar (unused-import)

-----------------------------------------------------------------------
Your code has been rated at -30.00/10 (previous run: -80.00/10, +50.00)
```

**Comment by hippo91** [comment]
@csymeonides-mf thanks for your report. From @bdegreve investigations (thanks by the way :+1) it seems that this is not a pylint bug.
I'm i wrong?
@kayabaNerve i think it will be clearer to open an new issue but maybe is it related to #1361 ?

**Comment by csymeonides-mf** [comment]
@bdegreve Apologies I forgot to reply to your comment. You're right, I hadn't installed the package, and that's because we use [pip-tools](https://pypi.org/project/pip-tools/) and there were some issues with installing local packages. So I agree with @hippo91, this isn't really a pylint bug.
